### PR TITLE
Add dodo optimism

### PR DESCRIPTION
**Pull request by owen05**
Brief comments on the purpose of your changes:

- Add dodo pools & trades on Optimism

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/dodo/optimism/dodo_optimism_sources.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          - name: contract_address
+            description: 'Address for the liquidity pool used in transaction'
+          - name: evt_tx_hash
+            description: 'Transaction hash of the event'
+          - name: evt_index
+            description: 'Event index'
+          - name: evt_block_time
+            description: 'Timestamp for block event time in UTC'
+
+      - name: DPP_evt_DODOSwap
```
Comment (Hosuke) [inline models/dodo/optimism/dodo_optimism_sources.yml:53]: Seems this source cannot be found:
https://dune.com/queries?d=1&blockchain=optimism&category=decoded_project&namespace=dodo

Comment (owen05) [reply models/dodo/optimism/dodo_optimism_sources.yml:53]: Have already decoded DPP contract on optimism. And fix the error

File: models/dodo/optimism/dodo_optimism_schema.yml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
+      - check_dex_seed:
+          blockchain: optimism
+          project: DODO
+          version: 1
```
Comment (Hosuke) [inline models/dodo/optimism/dodo_optimism_schema.yml:118]: By the way, which part in the sql is corresponding to DODO version 1?

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @owen05 

[Sample dodo optimism trades](https://dune.com/queries/2042470?d=1)

**Review by jeff-dude — APPROVED** [review]

File: models/dodo/optimism/dodo_optimism_schema.yml (hunk 2 above)
Comment (owen05) [reply models/dodo/optimism/dodo_optimism_schema.yml:118]: There is no version 1  yet. RD team will add soon

**Pull request merged**